Dockerfile-builder.template

```diff
@@ -340,10 +344,15 @@ RUN set -eux; \
 
 RUN set -eux; \
 	nproc="$(nproc)"; \
+# https://git.busybox.net/busybox/tree/Makefile?h=1_37_stable#n145
 {{ if env.variant == "uclibc" then ( -}}
 	CROSS_COMPILE="$(basename /usr/src/buildroot/output/host/usr/*-buildroot-linux-uclibc*)"; \
 	export CROSS_COMPILE="$CROSS_COMPILE-"; \
-{{ ) else "" end -}}
+{{ ) else ( -}}
+# we need to override SUBARCH explicitly (via ARCH) to avoid "uname -m" which gives the wrong answer for builds like i386 on an amd64 machine because kernel architecture != userspace architecture
+	ARCH="$(dpkg-architecture --query DEB_HOST_ARCH_CPU)"; \
```

Comment:
Technically, we could do this via maintaining a `case` statement off `dpkg --print-architecture` or `apk --print-arch`, but the end result is the same with more code for _us_ to maintain, so I went for the simpler answer (pulling in a tool that I know will DTRT, or at least close enough for our needs here).

Another option would be setting `CROSS_COMPILE` appropriately in _all_ cases (because even Debian versions natively can be invoked via the "cross" prefix), but it felt wrong to signal to the build system that we're "cross" building when it's ... not.

Reply:
Heh, I made the wrong call, but with a quick "finding out" turnaround: https://github.com/docker-library/busybox/pull/217